### PR TITLE
mypy and flake8

### PR DESCRIPTION
**Pull request by eggplants**
I followed flake8 and `mypy --strict` and fixed their errors. And also I introduced CI of flake8 and mypy. 

---

```shellsession
$ git checkout main
$ mypy --strict tests scripts budoux
Found 87 errors in 11 files (checked 15 source files)
$ flake8 | wc -l
331
```

```shellsession
$ git checkout mypy
$ mypy --strict tests scripts budoux
Success: no issues found in 14 source files
$ flake8 | wc -l
0
$ grep -r 'type: ignore' | grep -v .git
tests/test_train.py:from scripts import train  # type: ignore # noqa (module hack)
$ grep -r noqa | grep -v .git
tests/test_parser.py:from budoux import parser  # noqa (module hack)
tests/test_feature_extractor.py:from budoux import feature_extractor, utils  # noqa (module hack)
tests/test_train.py:from scripts import train  # type: ignore # noqa (module hack)
tests/test_main.py:from budoux import main  # noqa (module hack)
scripts/context.py:from budoux import Parser, feature_extractor, utils  # noqa (unused)
scripts/train.py:           == True) - 1  # noqa (cannot replace `==` with `is`)
```

### CODE REVIEW

**Comment by eggplants** [comment]
Type subscription in numpy (like `np.floating[typing.Any]`) is available on Python >=3.9.

File: scripts/train.py [hunk 1]
```diff
@@ -135,7 +147,7 @@ def fit(X: np.ndarray, Y: np.ndarray, features: typing.List[str], iters: int,
 
   for t in range(iters):
     print('=== %s ===' % (t))
-    res: np.ndarray = w.dot(Y_train[:, None] ^ X_train) / w.sum()
+    res: npt.NDArray[typing.Any] = w.dot(Y_train[:, None] ^ X_train) / w.sum()
```
Comment (tushuhei) [inline scripts/train.py:150]: Can we type this as `npt.NDArray[np.float64]`?

**Review by tushuhei — CHANGES_REQUESTED** [review]
Thank you for a lot of improvements to make our code typed! Let me ask some minor fixes.

File: budoux/feature_extractor.py [hunk 2]
```diff
@@ -138,12 +138,12 @@ def process(source_filename: str, entries_filename: str) -> None:
...
   """
   with open(source_filename, encoding=sys.getdefaultencoding()) as f:
-    datam = f.readlines()
+    data = f.readlines()
   with open(entries_filename, 'w', encoding=sys.getdefaultencoding()) as f:
     f.write('')
 
-  for data in datam:
-    chunks = data.strip().split(SEP)
+  for datam in data:
```
Comment (tushuhei) [inline budoux/feature_extractor.py:145]: nit: s/datam/datum

**Review by tushuhei — CHANGES_REQUESTED** [review]

**Review by tushuhei — APPROVED** [review]
LGTM. Thank you so much!

**Pull request merged**